File: app/src/main/java/de/slgdev/leoapp/view/LeoAppLayerActivity.java
package de.slgdev.leoapp.view;

import android.os.Bundle;
import android.support.annotation.CallSuper;
import android.support.annotation.IdRes;
import android.support.annotation.LayoutRes;
import android.support.annotation.StringRes;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.Toolbar;

import de.slgdev.leoapp.R;

public abstract class LeoAppLayerActivity extends ActionLogActivity {

    @Override
    protected void onCreate(Bundle b) {
        super.onCreate(b);
        setContentView(getContentView());
        initToolbar();
    }

    /**
     * Muss in der Implementation die Ressourcen-ID des Activity-Layouts zurückgeben.
     *
     * @return id des Activity-Layouts, zB. R.layout.startseite
     */
    protected abstract @LayoutRes
    int getContentView();

    /**
     * Soll die ID der Toolbar zurückgeben.
     *
     * @return Toolbar-ID
     */
    protected abstract @IdRes
    int getToolbarId();

    /**
     * Soll die String-Ressource des Titels der Toolbar zurückgeben.
     *
     * @return Text-ID, zb. R.string.title_main
     */
    protected abstract @StringRes
    int getToolbarTextId();

    /**
     * Allgemeine Methode zum Einrichten der Toolbar. Alle Änderungen wirken sich auf die gesamte App (NUR Feature-Toolbars - Keine der sonstigen Activites) aus.
     * Überschreibende Methoden müssen super.initToolbar() aufrufen.
     */
    @CallSuper
    protected void initToolbar() {
        Toolbar toolbar = findViewById(getToolbarId());
        toolbar.setTitleTextColor(ContextCompat.getColor(getApplicationContext(), android.R.color.white));
        toolbar.setTitle(getString(getToolbarTextId()));
        setSupportActionBar(toolbar);
        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_arrow_left);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }
}

File: core/src/main/java/de/leoappslg/core/activity/LeoAppLayerActivity.kt
@file:Suppress("unused", "MemberVisibilityCanBePrivate")

package de.leoappslg.core.activity

import android.os.Bundle
import android.view.MenuItem

import androidx.annotation.CallSuper
import androidx.annotation.IdRes
import androidx.annotation.LayoutRes
import androidx.annotation.StringRes
import androidx.appcompat.widget.Toolbar
import androidx.core.content.ContextCompat
import de.leoapp_slg.core.R
import de.leoappslg.core.activity.ActionLogActivity

abstract class LeoAppLayerActivity : ActionLogActivity() {

    override fun onCreate(b: Bundle?) {
        super.onCreate(b)
        setContentView(getContentView())
        initToolbar()
    }

    /**
     * Muss in der Implementation die Ressourcen-ID des Activity-Layouts zurückgeben.
     *
     * @return id des Activity-Layouts, zB. R.layout.startseite
     */
    @LayoutRes
    protected abstract fun getContentView(): Int

    /**
     * Soll die ID der Toolbar zurückgeben.
     *
     * @return Toolbar-ID
     */
    @IdRes
    protected abstract fun getToolbarId(): Int

    /**
     * Soll die String-Ressource des Titels der Toolbar zurückgeben.
     *
     * @return Text-ID, zb. R.string.title_main
     */
    @StringRes
    protected abstract fun getToolbarTextId(): Int

    /**
     * Allgemeine Methode zum Einrichten der Toolbar. Alle Änderungen wirken sich auf die gesamte App (NUR Feature-Toolbars - Keine der sonstigen Activites) aus.
     * Überschreibende Methoden müssen super.initToolbar() aufrufen.
     */
    @CallSuper
    protected fun initToolbar() {
        val toolbar: Toolbar = findViewById(getToolbarId())
        toolbar.setTitleTextColor(ContextCompat.getColor(applicationContext, android.R.color.white))
        toolbar.title = getString(getToolbarTextId())
        setSupportActionBar(toolbar)
        supportActionBar!!.setHomeButtonEnabled(true)
        supportActionBar!!.setHomeAsUpIndicator(R.drawable.ic_arrow_left)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    @CallSuper
    override fun onOptionsItemSelected(mi: MenuItem): Boolean {
        if (mi.itemId == android.R.id.home) {
            finish()
        }
        return true
    }
}